File: resources/expected/ros2/building_control_gen_mixed/lax/src/building_control_cpp_pkg/src/base_code/enum_converter.cpp
#include "building_control_cpp_pkg/base_headers/enum_converter.hpp"

const char* enumToString(building_control_cpp_pkg_interfaces::msg::TempUnit* value) {
    switch (value->temp_unit) {
        case building_control_cpp_pkg_interfaces::msg::TempUnit::TEMP_UNIT_FAHRENHEIT:
            return "TempUnit Fahrenheit";
        case building_control_cpp_pkg_interfaces::msg::TempUnit::TEMP_UNIT_CELSIUS:
            return "TempUnit Celsius";
        case building_control_cpp_pkg_interfaces::msg::TempUnit::TEMP_UNIT_KELVIN:
            return "TempUnit Kelvin";
        default:
            return "Unknown value for TempUnit";
    }
}

const char* enumToString(building_control_cpp_pkg_interfaces::msg::FanAck* value) {
    switch (value->fan_ack) {
        case building_control_cpp_pkg_interfaces::msg::FanAck::FAN_ACK_OK:
            return "FanAck Ok";
        case building_control_cpp_pkg_interfaces::msg::FanAck::FAN_ACK_ERROR:
            return "FanAck Error";
        default:
            return "Unknown value for FanAck";
    }
}

const char* enumToString(building_control_cpp_pkg_interfaces::msg::FanCmd* value) {
    switch (value->fan_cmd) {
        case building_control_cpp_pkg_interfaces::msg::FanCmd::FAN_CMD_ON:
            return "FanCmd On";
        case building_control_cpp_pkg_interfaces::msg::FanCmd::FAN_CMD_OFF:
            return "FanCmd Off";
        default:
            return "Unknown value for FanCmd";
    }
}

File: resources/expected/ros2/python-building_control_gen_mixed_inverted_topics/strict/src/building_control_py_pkg/building_control_py_pkg/base_code/enum_converter.py
#!/usr/bin/env python3
from building_control_py_pkg_interfaces.msg import TempUnit
from building_control_py_pkg_interfaces.msg import FanAck
from building_control_py_pkg_interfaces.msg import FanCmd

#========================================================
# Re-running Codegen will overwrite changes to this file
#========================================================

def enumToString(value):
    typedValue = TempUnit()
    typedValue.data = value
    match (typedValue.temp_unit):
        case TempUnit.TEMP_UNIT_FAHRENHEIT:
            return "TempUnit Fahrenheit"
        case TempUnit.TEMP_UNIT_CELSIUS:
            return "TempUnit Celsius"
        case TempUnit.TEMP_UNIT_KELVIN:
            return "TempUnit Kelvin"
        case default:
            return "Unknown value for TempUnit"

def enumToString(value):
    typedValue = FanAck()
    typedValue.data = value
    match (typedValue.fan_ack):
        case FanAck.FAN_ACK_OK:
            return "FanAck Ok"
        case FanAck.FAN_ACK_ERROR:
            return "FanAck Error"
        case default:
            return "Unknown value for FanAck"

def enumToString(value):
    typedValue = FanCmd()
    typedValue.data = value
    match (typedValue.fan_cmd):
        case FanCmd.FAN_CMD_ON:
            return "FanCmd On"
        case FanCmd.FAN_CMD_OFF:
            return "FanCmd Off"
        case default:
            return "Unknown value for FanCmd"

